File: 프로그래머스/2/42626. 더 맵게/더 맵게.py
import heapq
def solution(scoville, K):
    answer = 0
    heap = scoville
    heapq.heapify(heap)
    
    while len(heap) >= 2 and heap[0] < K:
        a = heapq.heappop(heap)
        b = heapq.heappop(heap)
        heapq.heappush(heap, a+b*2)
        answer += 1
    
    if heap[0] >= K: return answer
    return -1

File: 프로그래머스/2/42626. 더 맵게/더 맵게.java
import java.util.*;

class Solution {
    public int solution(int[] scoville, int K) {
        int answer = 0;
        
        PriorityQueue<Integer> pq = new PriorityQueue<>();
        for(int s :scoville) pq.offer(s);
        
        while(pq.peek() < K && pq.size() >= 2) {
            int a = pq.poll();
            int b = pq.poll();
            int mix = a + 2*b;
            pq.offer(mix);
            answer++;
        }
        return pq.peek() >= K ? answer : -1;
    }
}